src/metadata/metadata_superblock.c

```diff
@@ -148,6 +241,19 @@ static void ocf_metadata_load_superblock_post(ocf_pipeline_t pipeline,
 		OCF_PL_FINISH_RET(pipeline, -OCF_ERR_INVAL);
 	}
 
+	for_each_core_metadata(cache, core, core_id) {
+		muuid = ocf_metadata_get_core_uuid(cache, core_id);
+		uuid.data = muuid->data;
+		uuid.size = muuid->size;
+
+		volume_type = ocf_ctx_get_volume_type(cache->owner,
+				core->conf_meta->type);
+
+		/* Initialize core volume */
+		ocf_volume_init(&core->volume, volume_type, &uuid, false);
+		core->has_volume = true;
```

Comment:
this is already done in _ocf_mngt_load_add_cores, you have effectively duplicated this part.